### ISSUE TITLE
NotImplemented Error in solve

### ISSUE THREAD

**Issue by proy87** [sympy/sympy#16358]
When I try to solve `(-x + 1)/Abs(x - 1)`, where `x` is finite, I got the following `Exception`: `NotImplementedError: solving Abs(x - 1) when the argument is not real or imaginary.`

What assumptions should be made in order to solve it?

**Comment by oscarbenjamin** [comment]
Not sure. The numerator is only zero when x=1 but then the denominator is also zero so that's a singularity. What do you expect solve to return?

**Comment by proy87** [comment]
@oscarbenjamin , empty set, not Exception.

**Comment by proy87** [comment]
@oscarbenjamin `(-x + 2)/Abs(x - 1)` raises same `Exception`.

**Comment by oscarbenjamin** [comment]
You can get that with:
```julia
In [21]: x = Symbol('x', real=True)                                                                                                            

In [22]: solve((1-x)/(abs(x-1)), x)                                                                                                            
Out[22]: []
```
You can also use solveset:
```
In [24]: solveset((1-x)/(abs(x-1)), x, Reals)                                                                                                  
Out[24]: ∅
```

I don't think it should raise when it's in the denominator like this though. We should only need to takes roots from the numerator and check that they aren't also roots of the denominator.

**Comment by smichr** [comment]
> I don't think it should raise when it's in the denominator like this though

agreed - premature pessimism on the part of solve, perhaps checking too soon whether Abs is present.
